src/controllers/sendResetToken.js

```diff
@@ -0,0 +1,114 @@
+import { randomBytes } from 'crypto';
+import models from '../db/models';
+import sendgrid from '../middlewares/helper';
+
+const { User } = models;
+
+/**
+  * @description class representing reset user password
+  *
+  * @class SendResettoken
+  */
+class SendResetToken {
+  /**
+    * @description - This method is responsible for creating creating password reset token
+    *
+    * @static
+    * @param {object} request - Request sent to the router
+    * @param {object} response - Response sent from the controller
+    *
+    * @returns {object} - object representing response message
+    *
+    */
+  static async resetToken(request, response) {
+    try {
+      const { email } = request.body;
+      if (email.trim().length < 1) {
+        return response.status(400).json({
+          status: 'Fail',
+          message: 'The email field cannot be empty'
+        });
+      }
+      const userData = await User.findOne({
+        where: { email }
+      });
+
+      if (!userData) {
+        return response.status(400).json({
+          status: 'Fail',
+          message: 'An email has been sent to your mail',
+        });
+      }
+      const userToken = randomBytes(40).toString('hex');
+      const tokenExpiresIn = Date.now() + 3600000;
+
+      const { username, password } = userData.dataValues;
+      const userUpdatedData = await userData.update({
+        token: userToken,
+        username,
+        email,
+        password,
+        passwordExpiresIn: tokenExpiresIn || userData.passwordExpiresIn,
+      });
+      const { HOST_URL } = process.env;
+      await sendgrid(userData.email, 'noreply@authors.com', HOST_URL, userToken);
```

Comment:
@okoroemeka amongst the parameters you're passing to the sendgrid method, I can see you're passing a static email 'noreply@authors.com', may I know why you're taking this approach?
What if the whole email or domain changes, would we need to change it in the code as well?

Reply:
So what do you suggest I pass in there.? I would like to know what you think.